corehq/blobs/fsdb.py

```diff
@@ -121,9 +183,12 @@ def get_path(self, identifier=None, bucket=DEFAULT_BUCKET):
 def safejoin(root, subpath):
     """Join root to subpath ensuring that the result is actually inside root
     """
-    root = realpath(root)
-    if not SAFENAME.match(subpath):
+    if (subpath.startswith(("/", ".")) or
+            "/../" in subpath or
+            subpath.endswith("/..") or
+            not SAFENAME.match(subpath)):
         raise BadName("unsafe path name: %r" % subpath)
```

Comment:
replace this if with `check_safe_key`

Reply:
https://github.com/dimagi/commcare-hq/pull/21569/commits/94141189d0a66b901a7837c2480cf1c99313288d